go/vumitools/middleware.py

```diff
@@ -603,6 +604,64 @@ def handle_failure(self, failure, connector_name):
         return failure
 
 
+class ConversationMetricsMiddlewareConfig(MetricsMiddlewareConfig):
+
+    redis_manager = ConfigDict(
+        "Redis configuration parameters", default={}, static=True)
+
+
+class ConversationMetricsMiddleware(MetricsMiddleware):
+    """
+    Middleware that stores which conversations have received or sent messages
+
+    :param dict redis_manager:
+        Connection configuration details for Redis.
+    """
+
+    CONFIG_CLASS = ConversationMetricsMiddlewareConfig
+
+    @inlineCallbacks
+    def setup_middleware(self):
+        # We don't use a VumiApi here because we don't have a Riak config for
+        # it.
+        self.redis = yield TxRedisManager.from_config(
+            self.config.redis_manager)
```

Comment:
@justinvdm you suggested I use a sub_manager here and in the task but I need an instance of a manager to do that and I'm not sure what instance to use.